multicluster/controllers/multicluster/serviceexport_controller.go

```diff
@@ -172,29 +177,54 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			if err := cleanup(); err != nil {
 				return ctrl.Result{}, err
 			}
-			err = r.updateSvcExportStatus(ctx, req, notFound)
+			err = r.updateSvcExportStatus(ctx, req, serviceNotFound)
```

Comment:
I think you can update Status repeatedly if there are other Service/Endpoint changes happen, even they do not impact the Status value?

Maybe not a big deal, or we should save Status to installedSvcs? The same apply to cleanup() - should we keep the Service in installedSvcs, as long it has been processed, so we do not call cleanup() multiple times unnecessarily?

Reply:
I left this part unchanged considering there is no harm to update the status since we already skip watching ServiceExport's status change event, but I added a `ResourceVersionChangedPredicate` for Service and Endpoint to limit the mapping events. It will only get notified when Service or Endpoints resource version changes.
When the ServiceExport is not found, it will check the 'svcInstalled' before call cleanup, but for Service and Endpoint, I left it to call `cleanup()` anyway considering most events will be skipped if there is no existing ServiceExport.

Reply:
There is no big "harm", but it is unnecessary and in some cases may look strange that we keep updating Status for no reason or keep deleting non-existing resources (will it generate strange logs by kube-apiserver or our code?).

Anyway, at least we need not to fix it in this PR.

Reply:
I added a step to compare status difference before update the ServiceExport, so it can help to avoid unnecessary update.